### ISSUE TITLE
Launcher should appear on active monitor

### ISSUE THREAD

**Issue by tomaskikutis** [otsaloma/catapult#16]
```
$ catapult --version
catapult 0.6
```

```
$ lsb_release -a
No LSB modules are available.
Distributor ID:	Pop
Description:	Pop!_OS 22.04 LTS
Release:	22.04
Codename:	jammy
```

When I press ctrl+space, the launcher always appears on a secondary monitor which is my laptop's built-in display, but I have an external monitor set as primary display. Before clicking the shortcut I was using an application on the primary display.

**Comment by otsaloma** [comment]
Yeah, I haven't tested on multiple monitors, so it might not work right. I'm in the process of porting to GTK 4 and GDK 4 (#15) and it seems the whole concept of a primary monitor is gone from GDK. I can't really find good documentation on this, so it's hard to figure out.

**Comment by otsaloma** [comment]
Also, this might be one of those things are simply not doable anymore as GTK/GDK has dropped support for pretty much everything that's not possible on Wayland. It might be fixable in the current GTK 3 code, but I'm not going to work on that anymore. Some of the toolkit changes seem unfortunate to me, but no point trying to fight it.

**Comment by tomaskikutis** [comment]
I'm not familiar with GTK/GDK API, but I'm using pop os 22.04 and its launcher is able to show on the correct display. I have no idea how they did it, whether at OS level or userland but it does work.

**Comment by otsaloma** [comment]
Based on a quick reading, the whole Pop OS shell is a GNOME shell extension and the launcher part of that, so it's a whole different thing.
